src/Model/Item.php

```diff
@@ -1405,7 +1405,9 @@ public static function insert(array $item, int $notify = 0, bool $post_local = t
 			self::updateDisplayCache($posted_item['uri-id']);
 		}
 
-		Post\Engagement::storeFromItem($posted_item);
+		if ($inserted) {
```

Comment:
Is it the only statement that needs to be inside this condition?

Reply:
I haven't had a deep look at the code concerning this question.